code/datums/spell.dm

```diff
@@ -148,6 +147,8 @@ GLOBAL_LIST_INIT(spells, typesof(/obj/effect/proc_holder/spell))
 	var/datum/spell_handler/custom_handler
 	/// List with the handler datums per spell type. Key = src.type, value = the handler datum created by create_new_handler()
 	var/static/list/spell_handlers = list()
+	/// handles a given spells cooldowns. tracks the time until its off cooldown,
+	var/datum/spell_cooldown/cooldown_handler
```

Comment:
doesn't use a static like the others as it needs a reference to the parent spell.

Reply:
This is going to need deleting/nulling in Destroy. And come to think of it, none of the other vars are doing that either, which is bad. Like `custom_handler`, `targeting`, etc.

Reply:
`custom_handler` and `targeting` should never be qdeleted. Since they are singletons used by all other spells of the same spell type.

Reply:
On the non-static part. You can make it so that the `cooldown_handler` is called when the spell processes. And the spell gives itself as reference to the `cooldown_handler` which will use that to do the button updating etc.
Not sure if it is worth it though